### PR TITLE
Wip ganesha

### PR DESCRIPTION
**Pull request by theanalyst**

### CODE REVIEW

**Comment by theanalyst** [comment]
supersedes #54

File: srv/pillar/ceph/rgw.sls-example [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#rgw_configurations:
+#  rgw:
+#    users:
+#      - { name: "demo", display: "Demo", email: "demo@demo.nil" }
+#      - { name: "demo1", display: "Demo1", email: "demo1@demo.nil" }
+
+rgw_configurations:
+  silver:
+    users:
+      - { name: "demo", display: "Demo", email: "demo@demo.nil" }
```
Comment (theanalyst) [inline srv/pillar/ceph/rgw.sls-example]: we'll rename the dict as {uid: "demo", "name": "Demo", email: "foo@bar.com"}? so as to align more with rgw's own convention, also uid and email has to be unique, name not really

File: srv/salt/_modules/rgw.py [hunk 2]
```diff
@@ -13,8 +19,88 @@ def configurations():
...
+            return []
+    return []
+
+def add_users(pathname="/srv/salt/ceph/rgw/cache"):
+    """
+    Write each user to its own file
+    """
+    for role in __pillar__['rgw_configurations']:
+        for user in __pillar__['rgw_configurations'][role]['users']:
+            command = "radosgw-admin user create --uid={} --display-name={} --email={}".format(user['name'], user['display'], user['email'])
```
Comment (theanalyst) [inline srv/salt/_modules/rgw.py]: Need to modify this to take up additional and optional parameters like `email` `system` and `access` and `secret` flag, I'll rework this slightly

**Comment by supriti** [comment]
Wiki on how to deploy nfs-ganesha: https://github.com/SUSE/DeepSea/wiki/NFS-Ganesha

**Pull request merged**